### PR TITLE
implement AddZoneRecord

### PR DESCRIPTION
**Pull request by jonlil**

### CODE REVIEW

File: dns.go [hunk 1]
```diff
@@ -46,6 +46,41 @@ func (api *API) AddSubdomain(domain string, subdomain string) (*Status, error) {
...
+		return err
+	}
+
+	for _, element := range results {
+		// Exclude ID before equality check
+		id := element.ID
+		element.ID = 0
+
+		// Compare by value
+		if *&element == *record {
```
Comment (jonlil) [inline dns.go]: is this bad practice? @akarl818

Comment (akarl) [reply dns.go]: Looks fine. Although wouldn't `element == *record` be the same thing?

File: dns.go [hunk 2]
```diff
@@ -46,6 +46,41 @@ func (api *API) AddSubdomain(domain string, subdomain string) (*Status, error) {
...
+	if err := api.XMLRPCClient().Call("addZoneRecord", args, &result); err != nil || result != "OK" {
+		return err
+	}
+
+	results, err := api.GetZoneRecords(domain, subdomain)
+	if err != nil {
+		return err
+	}
+
+	for _, element := range results {
```
Comment (akarl) [inline dns.go:71]: Are you sure there isn't any other way to accomplish what you need this loop for? The things with the IDs seems a little hacky?

**Review by akarl — APPROVED** [review]

File: dns.go (hunk 2 above)
Comment (jonlil) [reply dns.go:71]: I'm not aware of any alternative way. The problem is that their API isn't returning any record_id in the addZoneRecord so the alternative I have found is this hack! Documentation should be improved.

The reason for doing this is to be able to keep track of the record in an internal system

File: dns.go (hunk 1 above)
Comment (jonlil) [reply dns.go]: That looks a lot simpler! Thanks for the suggestion

File: dns.go (hunk 2 above)
Comment (jonlil) [reply dns.go:71]: solved in fe7d6ca

**Pull request merged**